test/unit/awssigv4signer.test.js

```diff
@@ -0,0 +1,407 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+const { test } = require('tap');
+const { URL } = require('url');
+const { v4: uuidv4 } = require('uuid');
+const AwsSigv4Signer = require('../../lib/aws/AwsSigv4Signer');
+const AwsSigv4SignerError = require('../../lib/aws/errors');
+const { Connection } = require('../../index');
+const { Client, buildServer } = require('../utils');
+
+test('Sign with SigV4', (t) => {
```

Comment:
Add a test for the absolute minimal syntax, aka with default signer options/credentials.

Reply:
I think that would be `test('Basic aws (promises)', (t) => {` line 96 and `test('Basic aws (callback)', (t) => {` line 288
those are using the basic options/credentials

Reply:
Yes, this would be the basic case. I agree.